libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -326,24 +326,12 @@ namespace eosio { namespace chain {
 
          static std::optional<uint64_t> convert_exception_to_error_code( const fc::exception& e );
 
-         signal<void(uint32_t)>                        block_start; // block_num
-         signal<void(const signed_block_ptr&)>         pre_accepted_block;
-         signal<void(const block_state_legacy_ptr&)>   accepted_block_header;
-         signal<void(const block_state_legacy_ptr&)>   accepted_block;
-         signal<void(const block_state_legacy_ptr&)>   irreversible_block;
+         signal<void(uint32_t)>                        block_start;
+         signal<void(std::tuple<const signed_block_ptr&, const block_id_type&, const signed_block_header&, uint32_t>)> accepted_block_header;
+         signal<void(std::tuple<const signed_block_ptr&, const block_id_type&, const signed_block_header&, uint32_t>)> accepted_block;
+         signal<void(std::tuple<const signed_block_ptr&, const block_id_type&, uint32_t>)> irreversible_block;
```

Comment:
Why the tuples instead of passing a list of arguments?

Reply:
Never mind, got it, looks like it is a restriction from the `channels` of `plugin_interface`.